spec/request/messages/index_spec.rb

```diff
@@ -0,0 +1,48 @@
+require 'rails_helper'
+
+describe 'GET #index messages', type: :request do
+  include_context 'conversation_between_two_users'
+
+  let!(:messages) do
+    create_list(:message, 21, user: user, conversation: conversation)
```

Comment:
Wow, 21 is a lot. When possible, create fewer objects so you can decrease the time for all the tests.
Can't you modify the page number for the test?